hudi-common/src/test/java/org/apache/hudi/common/table/read/TestKeepValuesPartialMergingUtils.java

```diff
@@ -0,0 +1,334 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.common.table.read;
+
+import org.apache.hudi.common.engine.HoodieReaderContext;
+import org.apache.hudi.common.util.collection.Pair;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.IndexedRecord;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class TestKeepValuesPartialMergingUtils {
+  private KeepValuesPartialMergingUtils<IndexedRecord> keepValuesPartialMergingUtils;
+  private HoodieReaderContext<IndexedRecord> mockReaderContext;
+  private Schema fullSchema;
+  private Schema partialSchema;
+  private Schema readerSchema;
+
+  @BeforeEach
+  void setUp() {
+    keepValuesPartialMergingUtils = new KeepValuesPartialMergingUtils<>();
+    mockReaderContext = mock(HoodieReaderContext.class);
+
+    // Create test schemas
+    fullSchema = Schema.createRecord("TestRecord", "Test record", "test", false);
+    fullSchema.setFields(Arrays.asList(
+        new Schema.Field("id", Schema.create(Schema.Type.STRING), "ID field", null),
+        new Schema.Field("name", Schema.create(Schema.Type.STRING), "Name field", null),
+        new Schema.Field("age", Schema.create(Schema.Type.INT), "Age field", null),
+        new Schema.Field("city", Schema.create(Schema.Type.STRING), "City field", null)
+    ));
+
+    partialSchema = Schema.createRecord("TestRecord", "Test record", "test", false);
+    partialSchema.setFields(Arrays.asList(
+        new Schema.Field("id", Schema.create(Schema.Type.STRING), "ID field", null),
+        new Schema.Field("name", Schema.create(Schema.Type.STRING), "Name field", null)
+    ));
+
+    readerSchema = Schema.createRecord("TestRecord", "Test record", "test", false);
```

Comment:
`readerSchema` is the same as `fullSchema`. I think you should have another schema that is different so the case where new, old, and reader schemas are all different.